capstonebundle/plugin/mos65xx/mos65xx.cpp

```diff
@@ -0,0 +1,35 @@
+// mos65xx.cpp
+#include "mos65xx.h"
+
+MOS65XX::MOS65XX(RDContext* ctx, cs_mode mode): Capstone(ctx, CS_ARCH_MOS65XX, mode) { }
+
+void MOS65XX::emulate(RDEmulateResult* result)
+{
+    rd_address address = RDEmulateResult_GetAddress(result);
+    auto* insn = this->decode(address, RDEmulateResult_GetView(result));
+    if(!insn) return;
+
+    // Instruction is decoded, you can use Capstone API to analyze it
+    
+    RDContext_SetAddressAssembler(m_context, address, this->endianness() == Endianness_Big ? MOS65XXBE_ID : MOS65XXLE_ID);
```

Comment:
This is used only in ARM, afaik is not neeed for MOS65XX, because it provides a single instruction set.

Tell REDasm to increase the view to the next possible instruction by calling
```cpp
RDEmulateResult_SetSize(result, insn->size); // Next time "emulate" is called is after insn->size bytes
```

After that you need to classify instructions in order to have a detailed listing (more case you handle and more detailed will become), see here (it just a big switch case statement): https://github.com/REDasmOrg/REDasm-Assemblers/blob/db0d698029d6776f4b9f3612498439621b473987/capstonebundle/plugin/arm32/common.cpp

The easiest way is to begin to catch return, call and jump statement